From 31dec0419e2f319ec0933fac5aa30074e9522892 Mon Sep 17 00:00:00 2001
From: Tor Vic <torvic9@mailbox.org>
Date: Thu, 7 Apr 2022 14:54:10 +0200
Subject: [PATCH 9/9] sched/alt: [Sync] sched/uclamp: Add dummy functions to
 sched_alt.h

Syncup of the following 5.17.2 commits:
  sched/uclamp: Fix iowait boost escaping uclamp restriction
  sched/sugov: Ignore 'busy' filter when rq is capped by uclamp_max

Link: https://gitlab.com/alfredchen/linux-prjc/-/issues/56
---
 kernel/sched/alt_sched.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/kernel/sched/alt_sched.h b/kernel/sched/alt_sched.h
index 6ff979a29..77cf57a11 100644
--- a/kernel/sched/alt_sched.h
+++ b/kernel/sched/alt_sched.h
@@ -649,6 +649,20 @@ static inline int sched_numa_find_closest(const struct cpumask *cpus, int cpu)
 }
 #endif
 
+static inline
+unsigned long uclamp_rq_util_with(struct rq *rq, unsigned long boost,
+				  struct task_struct *p)
+{
+	return boost;
+}
+
+static inline bool uclamp_rq_is_capped(struct rq *rq) { return false; }
+
+static inline bool uclamp_is_used(void)
+{
+	return false;
+}
+
 extern void swake_up_all_locked(struct swait_queue_head *q);
 extern void __prepare_to_swait(struct swait_queue_head *q, struct swait_queue *wait);
 
-- 
2.35.1.677.gabf474a5dd

